open-metadata-conformance-suite/open-metadata-conformance-suite-server/src/main/java/org/odpi/openmetadata/conformance/tests/repository/instances/TestSupportedEntityProxyLifecycle.java

```diff
@@ -1371,7 +1371,7 @@ public void cleanup() throws Exception
          *   createdRelationshipsTUT
          */
 
-        if (createdRelationshipsTUT != null && !createdRelationshipsTUT.isEmpty()) {
+        if (!createdRelationshipsTUT.isEmpty()) {
```

Comment:
This can be seen in the sonar report at https://sonarcloud.io/project/issues?fileUuids=AW-QewLl1UqzkPNMUrDy&resolved=false&id=odpi_egeria&open=AXUwjvDz8Z59dRxV-JY9

As I noticed myself, this does have to be null , and so sonar is reporting this as a 'code smell'

This is explained at https://sonarcloud.io/organizations/odpi/rules?open=java%3AS2589&rule_key=java%3AS2589

On that basis I agree with the analysis, and therefore with the technical aspects of the change. In some ways it clarifies behaviour, and simplifies the reading of the code.

I think this may be worthy of wider discussion, perhaps posting an example on slack. I wonder also how important the 'code smell' is. We should certainly review, but do we need to action? If we decided not to, we could create a document listing rules we've decided to not adopt (but therein we may get issues with other scanners). Do our other  scans show issues with this logic, or is it unique to sonar?

Reply:
@planetf1 In this case it doesn't feel that complicated, review the scans with the changes (or even review a few that are visible in the intellij plugin. In all but 1 case, each alert was obvious and just plainly wrong. 

These are being reported collectively as as a  classification  of security vulnerability  referred to as 'Insecure Configuration"

Reply:
I agree your proposal is correct. I'd probably myself have writing code following this anti pattern, so it was a good challenge. I'm ok with the change, especially as it will help us get to a much smaller list of issues, then we'll also be able to see much more easily any identified code flows where a change might mean a check is then needed